src/locator/content-claims-client.js

```diff
@@ -0,0 +1,193 @@
+import * as ed25519 from '@ucanto/principal/ed25519'
+import { base58btc } from 'multiformats/bases/base58'
+import { NotFoundError } from '../lib.js'
+import { from } from '@storacha/indexing-service-client/query-result'
+
+/**
+ * @import * as API from '../api.js'
+ * @import {Claim, IndexingServiceClient, Query, QueryError, QueryOk, Result, Kind} from '@storacha/indexing-service-client/api'
+ * @import {KnownClaimTypes, LocationClaim} from '@web3-storage/content-claims/client/api'
+ */
+
+/**
+ * @typedef {{ serviceURL?: URL, carpark?: import('@cloudflare/workers-types').R2Bucket, carparkPublicBucketURL?: URL}} LocatorOptions
+ */
+
+/**
+ * ContentClaimsClient mimics the indexing service client using the content claims service
+ * @implements {IndexingServiceClient}
+ */
+export class ContentClaimsClient {
+  /**
+   * @type {DigestMap<API.MultihashDigest, boolean>}
+   */
+  #indexCids
+
+  /**
+   * @type {URL|undefined}
+   */
+  #serviceURL
+
+  /**
+   * @type {import('@cloudflare/workers-types').R2Bucket|undefined}
+   */
+  #carpark
+
+  /**
+   * @type {URL | Undefined}
+   */
+  #carparkPublicBucketURL
+
+  /**
+   * @type {ed25519.EdSigner | undefined}
+   */
+  #signer
+
+  /**
+   * @param {LocatorOptions} [options]
+   */
+  constructor (options) {
+    this.#indexCids = new DigestMap()
+    this.#serviceURL = options?.serviceURL
+    this.#carpark = options?.carpark
+    this.#carparkPublicBucketURL = options?.carparkPublicBucketURL
+  }
+
+  async #getSigner () {
+    if (!this.#signer) {
+      this.#signer = await ed25519.Signer.generate()
+    }
+    return this.#signer
+  }
+
+  /**
+   *
+   * @param {Query} q
+   * @returns {Promise<Result<QueryOk, QueryError>>}
+   */
+  async queryClaims (q) {
+    /** @type {Claim[]} */
+    const claims = []
+    /** @type {Map<string, import('@web3-storage/blob-index/types').ShardedDAGIndexView>} */
+    const indexes = new Map()
+    const kind = q.kind || 'standard'
+    for (const digest of q.hashes) {
+      const digestClaims = (await Claims.read(digest, { serviceURL: this.#serviceURL })).filter((claim) => allowedClaimTypes[kind].includes(claim.type))
+      let indexBytes
+      if (digestClaims.length === 0) {
+        const backups = await this.#carparkBackup(digest)
+        if (backups) {
+          claims.push(backups.claim)
+          indexBytes = backups.indexBytes
+        }
+      } else {
+        claims.push(...digestClaims)
+        for (const claim of digestClaims) {
+          if (claim.type === 'assert/index') {
+            this.#indexCids.set(claim.index.multihash, true)
+          }
+          if (claim.type === 'assert/location' && this.#indexCids.has(Claims.contentMultihash(claim))) {
+            try {
+              const fetchRes = await fetchIndex(claim)
+              indexBytes = await fetchRes.bytes()
+            } catch (err) {
+              console.warn('unable to fetch index', err instanceof Error ? err.message : 'unknown error')
```

Comment:
"unknown error" is not useful, just log it and let console figure out formatting!
```suggestion
              console.warn('unable to fetch index', err)
```